src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -655,7 +657,9 @@ private AssetData GetManifestAsAsset(IImmutableList<AssetData> assets, string lo
                             Branch = GetAzDevBranch(),
                             Commit = GetAzDevCommit(),
                             IsStable = IsStableBuild.ToString(),
-                            PublishingVersion = manifestBuildData.PublishingVersion.ToString()
+                            PublishingVersion = (PublishingInfraVersion)manifestBuildData.PublishingVersion,
```

Comment:
Can you explain why this cast is now required?

Reply:
For the older MicrosoftDotNetVersionToolsVersion PublishingVersion was int, the newer MicrosoftDotNetVersionToolsVersion 
PublishingVersion is an enum.

Reply:
Should we be concerned that someone passes an invalid number here? the direct cast won't fail if provided an invalid int.

Reply:
That part of validation is done in arcade, since we are just passing the value I think we should be good. 

https://github.com/dotnet/arcade/blob/ba32978d76e590e1a5cde6b754eb4e9b8d212c73/src/Microsoft.DotNet.VersionTools/lib/src/BuildManifest/Model/BuildIdentity.cs#L90